src/main/java/com/urbanairship/datacube/dbharnesses/HbaseBatchIncrementer.java

```diff
@@ -0,0 +1,248 @@
+     *
+     * @return A map from the serialized {@link Address} to the bytes in the column after completion of the operation.
+     */
+    private Map<BoxedByteArray, byte[]> processBatchCallresults(List<Map.Entry<BoxedByteArray, T>> entries, Object[] objects) {
+        Map<BoxedByteArray, byte[]> successes = new HashMap<>();
+        for (int i = 0; i < objects.length; ++i) {
+            if (objects[i] != null && objects[i] instanceof Result) {
+                Result result = (Result) objects[i];
+                byte[] value = result.getValue(configuration.cf, HBaseDbHarness.QUALIFIER);
+                successes.put(entries.get(i).getKey(), value);
+            }
+        }
+        return successes;
+    }
+
+    static class CubeIncrementResults {
+        final IOException ioException;
+        final InterruptedException interrupt;
+        final Map<BoxedByteArray, byte[]> successes;
+
+        CubeIncrementResults(IOException ioException, InterruptedException interrupt, Map<BoxedByteArray, byte[]> successes) {
+            this.ioException = ioException;
+            this.interrupt = interrupt;
+            this.successes = successes;
+        }
+
+        public static Builder newBuilder() {
+            return new Builder();
+        }
+
+        public static class Builder {
+            IOException ioException;
+            InterruptedException interrupt;
+            Map<BoxedByteArray, byte[]> successes = new HashMap<>();
+
+            public Builder setIoException(IOException ioException) {
+                this.ioException = ioException;
+                return this;
+            }
+
+            public Builder setInterrupt(InterruptedException interrupt) {
+                this.interrupt = interrupt;
+                return this;
+            }
+
+            public Builder setSuccesses(Map<BoxedByteArray, byte[]> successes) {
+                this.successes = successes;
+                return this;
+            }
+
+            public CubeIncrementResults build() {
+                return new CubeIncrementResults(ioException, interrupt, ImmutableMap.copyOf(successes));
+            }
+
+            public CubeIncrementResults.Builder add(CubeIncrementResults cubeIncrementResults) {
+                setInterrupt(cubeIncrementResults.interrupt);
+                setIoException(cubeIncrementResults.ioException);
+                setSuccesses(ImmutableMap.<BoxedByteArray, byte[]>builder()
+                        .putAll(successes)
+                        .putAll(cubeIncrementResults.successes)
+                        .build());
+                return this;
+            }
+        }
+    }
+
+
+    private class IncrementerMetrics {
+        private final Histogram incrementSize;
+        private final Histogram incrementFailuresPerFlush;
+        private final Histogram batchesPerFlushHisto;
+        private final Timer batchWritesTimer;
+
+        public IncrementerMetrics(Class clazz, String metricsScope) {
```

Comment:
passing clazz as a parameter is goofy, but I wasn't prepared to break any metrics/recreate dashboards.